Client/Pages/BookPages/BookDetails.razor

```diff
@@ -1,63 +1,105 @@
+                    <td>@book.BookId</td>
+                    <td>
+                        <InputSelect @bind-Value="book.BookPublisherId" title="Publisher name">
+                            @foreach (var publisher in publishers)
+                                {
+                                <option value="@publisher.PublisherId">@publisher.PublisherName</option>
+                                }
+                        </InputSelect>
+                    </td>
+                    <td>
+                        @foreach (var author in authors)
                         {
-                            <option value="@publisher.PublisherId">@publisher.PublisherName</option>
+                            <div>
+                                <NavLink class="button" href=@($"/authorView/{author.AuthorId}")>@author.FullName</NavLink> 
+                                <button type="button" class="button" @onclick="(() => DeleteAuthor(author.AuthorId))">Delete author</button>
+                        </div>
                         }
-                    </select>
-                </td>
-                <td>Authors</td>
-                <td><button @onclick="Edit">Edit book</button></td>
-            </tr>
-        </tbody>
-    </table>
+                    <div>
+                        <InputSelect @bind-Value="bookAuthor.AuthorId">
+                            @foreach (var author in allAuthors)
+                            {
+                                    @if (!author.Books.Any(b => b.BookId.Equals(id)))
+                                    {
+                                <option value="@author.AuthorId">@author.FullName</option>
+                                        }
+                                }
+                        </InputSelect>
+                        <button type="button" class="button" @onclick="AddAuthor">Add author</button>
+                    </div>
+                    </td>
+                    <td><button type="submit">Edit book</button></td>
+                </tr>
+            </tbody>
+
+        </table>
+    </EditForm>
+    
 }
 @code {
-    [Parameter]
+[Parameter]
     public Guid id { get; set; }
-    public Test.Shared.Book book { get; set; } = null;
-    private string nameToEdit;
-    private Guid publisherId;
+    public Test.Shared.Book book { get; set; } = new Test.Shared.Book();
+    private List<Test.Shared.Publisher> publishers = new List<Test.Shared.Publisher>();
+    private ICollection<Author> authors { get; set; } = new List<Author>();
+    public List<Author> allAuthors { get; set; } = new List<Author>();
+    public BookAuthor bookAuthor { get; set; } = new BookAuthor();
 
     private async Task Edit()
     {
-        book.BookName = nameToEdit;
-        book.BookPublisherId = publisherId;
-        await Http.PutAsJsonAsync<Test.Shared.Book>("book", book);
+        await Http.PutAsJsonAsync<Test.Shared.Book>($"api/books/{id}", book);
```

Comment:
Same here as below. PUT should return the updated value that can be used to update the local list that automatically triggers re-rendering on state change.

The whole concept should be that if the server already gives you latest information, this can be merged on client side to avoid unnecessary queries for list objects.